src/transaction.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import type { BigNumber } from "bignumber.js";
+import type { Account, Transaction } from "./types";
+import type { AccountBridge } from "./bridge/types";
+
+export async function getTransactionStatus(
+  bridge: AccountBridge<any>,
+  account: Account,
+  transaction: Transaction
+): Promise<{
+  showFeeWarning: boolean,
+  estimatedFees: BigNumber,
+  amount: BigNumber,
+  totalSpent: BigNumber,
+  useAllAmount: boolean,
+  maxAmount: ?BigNumber
+}> {
+  const totalSpent = await bridge.getTotalSpent(account, transaction);
+  const amount = bridge.getTransactionAmount(account, transaction);
+  const useAllAmount = bridge.getTransactionExtra(
+    account,
+    transaction,
+    "useAllAmount"
+  );
+
+  let maxAmount;
+  if (useAllAmount) maxAmount = await bridge.getMaxAmount(account, transaction);
+
+  const estimatedFees = totalSpent - (maxAmount || amount);
```

Comment:
unfortunately this is not true for ERC20.

the fees is not `totalSpent minus amount` because this is basically zero for erc20.

sounds like we need a bridge.getEstimatedFees. (or we need this whole function to be in bridge)